fabric-particles-v1/src/main/java/net/fabricmc/fabric/impl/client/particle/FabricParticleManager.java

```diff
@@ -81,19 +90,38 @@ public boolean loadParticle(ResourceManager manager, Identifier id) {
 				throw new IllegalStateException("(Fabric) Missing texture list for particle " + id);
 			}
 
-			provider.setSprites(spriteIds);
+			unloadedSprites.put(id, spriteIds.stream()
+					.map(sprite -> new Identifier(sprite.getNamespace(), "particle/" + sprite.getPath()))
+					.collect(Collectors.toList())
+			);
 		} catch (IOException e) {
-			throw new IllegalStateException("Failed to load description for particle " + id, e);
+			throw new IllegalStateException("(Fabric) Failed to load description for particle " + id, e);
```

Comment:
No need; the stacktrace suffices

Reply:
I added this because the stack trace wasn't showing for our code. In fact, even if this method throws, it seemed like execution still went through to Mojang's original method.

Reply:
The cause (IOException e) definitely has a mojang stacktrace. Just look at the stack trace of the Illegal State.

Reply:
That's what I mean. It seems like the exception thrown in this method is swallowed somewhere by the Mixin boilerplate.

Reply:
then add prefixes like `fabric$` to the injection/redirect methods in mixins.

Reply:
That doesn't solve the problem, though? Exceptions aren't thrown at all.